src/app/g3w-ol/controls/streetviewcontrol.js

```diff
@@ -87,78 +126,121 @@ proto.setProjection = function(projection) {
+};
+
+/**
+ * @since v3.8
+ * 
+ * @listens google.maps.StreetViewPanorama#position_changed
+ * @listens google.maps.StreetViewPanorama#pov_changed
+ */
+proto.processStreetViewServiceData = function(data, status) {
+  // if (!this._panorama) { // TODO ?
+    this._panorama = new google.maps.StreetViewPanorama( document.getElementById('streetview'), { imageDateControl: true });
+  // }  
+
+  this._panorama.addListener('position_changed', this.onPanoramaPositionChanged.bind(this));
+  this._panorama.addListener('pov_changed',      this.onPanoramaPovChanged.bind(this));
+
+  if (data && data.location) {
+    this._panorama.setPosition(data.location.latLng);
     /**
-     * Listen on position change
-    */
-    self._panorama.addListener('position_changed', function() {
-      if (self.isToggled()) {
-        const lnglat = ol.proj.transform([this.getPosition().lng(), this.getPosition().lat()], 'EPSG:4326', self._projection.getCode());
-        self._streetViewFeature.setGeometry(
-          new ol.geom.Point(lnglat)
-        );
-        pixel = self._map.getPixelFromCoordinate(lnglat);
-        if ((pixel[0] + 15) > self._map.getSize()[0] || (pixel[1] + 15) > self._map.getSize()[1] || pixel[0] < 15 || pixel [1] < 15 ) {
-          self._map.getView().setCenter(lnglat);
-        }
-      }
-    });
-    if (data && data.location) {
-      self._panorama.setPov({
-        pitch: 0,
-        heading: 0
-      });
+     * @TODO compute intial heading evaluating it between the "openlayers" and "streetview" coordinates
+     */
+    // this._panorama.setPov({
+    //   heading: google.maps.geometry.spherical.computeHeading(data.location.latLng, new google.maps.LatLng(this._clickLat, this._clickLng)),
+    //   pitch: 0,
+    //   zoom: 0
+    // });
+  }
+};
 
-      self._panorama.setPosition(data.location.latLng);
-    }
-  }).then(response => {
-    if (response === undefined) {
-      GUI.closeContent();
-    }
-  }).catch(() => this.toggle())
+/**
+ * @since v3.8
+ */
+proto.onPanoramaPositionChanged = function() {
+
+  const pos = this._panorama.getPosition();
+
+  if (!pos || !this.isToggled()) {
+    return;
+  }
+
+  const lnglat = ol.proj.transform([ pos.lng(), pos.lat() ], 'EPSG:4326', this._projection.getCode());
+
+  this._streetViewFeature.setGeometry( new ol.geom.Point(lnglat) );
+
+  let pixel = this._map.getPixelFromCoordinate(lnglat);
+
+  // recenter map if marker is outside map bounds (15 = pixel padding)
+  if (
+    pixel[0] > this._map.getSize()[0] - 15 ||
+    pixel[1] > this._map.getSize()[1] - 15 ||
+    pixel[0] < 15 ||
+    pixel[1] < 15
+    ) {
+    this._map.getView().setCenter(lnglat);
+  }
+
+};
+
+/**
+ * @since v3.8
+ */
+proto.onPanoramaPovChanged = function() {
+  const pov = this._panorama.getPov();
+  this._iconRotation = toRadians(pov.heading + iconOffset);
+  /**
+   * @TODO really ugly, find out a simpler way to just update the `ol.style.Icon~rotation` value
+   */
+  this.onPanoramaPositionChanged();
```

Comment:
take a look at it before merging

Reply:
@Raruto need to be close https://github.com/g3w-suite/g3w-client/pull/338 before it